app/controllers/users_controller.rb

```diff
@@ -0,0 +1,30 @@
+class UsersController < ApplicationController
+  
+  def show
+    @user = User.find_by(id: params[:id])
+      
+    @upcoming_events = current_user.upcoming_events
+    @previous_events = current_user.previous_events
+  end
+
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      flash[:info] = "Account Created Successfully"
+      log_in(@user)
+      redirect_to @user
```

Comment:
Why not have the `redirect_to user` line in the `log_in` method in the `sessions_helper`?

Reply:
it's not every time you might want to redirect users to profile page after login that's why the redirect isn't in login